pkg/controllers/provisioning/v1alpha1/allocator/bind.go

```diff
@@ -0,0 +1,90 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package allocator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const KarpenterNodeLabelKey = "karpenter.sh/provisioner"
+
+type Binder struct {
+	kubeClient   client.Client
+	coreV1Client *corev1.CoreV1Client
+}
+
+func (b *Binder) Bind(ctx context.Context, provisioner *v1alpha1.Provisioner, node *v1.Node, pods []*v1.Pod) error {
+	// 1. Decorate node
+	b.decorate(provisioner, node)
+
+	// 2. Create node
+	if err := b.create(ctx, node); err != nil {
+		return err
+	}
+
+	// 3. Bind pods
+	for _, pod := range pods {
+		if err := b.bind(ctx, node, pod); err != nil {
+			zap.S().Errorf("Continuing after failing to bind, %s", err.Error())
+		} else {
+			zap.S().Debugf("Successfully bound pod %s/%s to node %s", pod.Namespace, pod.Name, node.Name)
+		}
+	}
+	return nil
+}
+
+func (b *Binder) decorate(provisioner *v1alpha1.Provisioner, node *v1.Node) {
+	node.ObjectMeta.Labels = map[string]string{
+		KarpenterNodeLabelKey: fmt.Sprintf("%s-%s", provisioner.Name, provisioner.Namespace),
```

Comment:
What do you think about using
`KarpenterNodeLabelKey: fmt.Sprintf("Karpenter.%s.%s", provisioner.Name, provisioner.Namespace),`

since other labels use periods in between and are prefixed for where they come from?